lib/reek/smells/boolean_parameter.rb

```diff
@@ -1,4 +1,5 @@
 require_relative 'smell_detector'
+require_relative 'smell_warning'
```

Comment:
Ok, I read up on your arguments above.

>> we can easily expose SmellWarning through a method (e.g. create_smell_warning).

Doesn't feel to good to me.
Quite frankly, I believe we should have gone down the road of smells/smell_types/.
With SmellDetector requiring SmellConfiguration.

But anyways.:)

The arguments from above are valid for sure. But putting this require in EVERY smell just seems wrong. We should put it in the SmellDetector. Practical usage over architectural correctness.:)